papis/commands/mv.py

```diff
@@ -55,58 +105,92 @@ def run(document: papis.document.Document,
-                                                       sort_field,
-                                                       sort_reverse)
+    documents = papis.cli.handle_doc_folder_query_all_sort(query,
+                                                           doc_folder,
+                                                           sort_field,
+                                                           sort_reverse,
+                                                           _all
+                                                           )
     if not documents:
         logger.warning(papis.strings.no_documents_retrieved_message)
         return
 
-    document = documents[0]
-
     lib_dir = os.path.expanduser(papis.config.get_lib_dirs()[0])
-
     completer = prompt_toolkit.completion.PathCompleter(
         only_directories=True,
-        get_paths=lambda: [lib_dir]
+        get_paths=lambda: [str(lib_dir)]
     )
 
-    try:
-        new_folder = os.path.join(
-            lib_dir,
-            prompt_toolkit.prompt(
-                message=(
-                    "Enter directory  : (Tab completion enabled)\n"
-                    "Current directory: ({dir})\n".format(
-                        dir=document.get_main_folder_name()
-                    ) + ">  "
-                ),
+    # If the user hasn't passed ``--folder_name``, we'll prompt for one that
+    # will be used throughout the process.
+    dest_path = None
+    if folder_name is None:
+        try:
+            subfolder = prompt_toolkit.prompt(
+                message="Destination folder: (Tab completion enabled)\n> ",
                 completer=completer,
-                complete_while_typing=True
-            ))
-    except Exception as exc:
-        logger.error("Failed to choose directory.", exc_info=exc)
-        return
-
-    logger.info("New document folder: '%s'.", new_folder)
-
-    if not os.path.exists(new_folder):
-        logger.info("Creating path '%s'.", new_folder)
-        os.makedirs(new_folder, mode=papis.config.getint("dir-umask") or 0o666)
-
-    run(document, new_folder, git=git)
+                complete_while_typing=True)
+            dest_path = os.path.join(lib_dir, subfolder)
+        except Exception as exc:
+            logger.error("Failed to choose directory.", exc_info=exc)
+            return
+
+    if not git and os.path.exists(os.path.join(lib_dir, ".git")):
+        git = papis.tui.utils.confirm("Add the changes to git?")
+
+    moves = []
+    for document in documents:
+        # If the user has passed ``--folder-name``, use that to generate the subfolder
+        if folder_name is not None:
+            subfolder = papis.format.format(folder_name, document)
+            dest_path = os.path.join(lib_dir, subfolder)
+
+        # This should never be reached, but just in case.
+        if dest_path is None:
+            logger.error("Failed to construct a folder path for the document.")
+            return
```

Comment:
```suggestion
        if not dest_path:
            logger.error("Failed to construct a folder path for the document: %s",
            			 papis.document.describe(document))
            continue
```
? To just skip the document? 

I'm never sure how to handle this, e.g. the update command will exit right away if it fails to update some key in some document, but most other commands will skip it and continue.